mlrun/projects/project.py

```diff
@@ -2003,20 +2019,39 @@ def create_remote(self, url, name="origin", branch=None):
         self.spec._source = self.spec.source or url
         self.spec.origin_url = self.spec.origin_url or url
 
-    def push(self, branch, message=None, update=True, remote=None, add: list = None):
+    def push(
+        self,
+        branch,
+        message=None,
+        update=True,
+        remote: str = None,
+        add: list = None,
+        author_name: str = None,
+        author_email: str = None,
+        secrets: Union[SecretsStore, dict] = None,
+    ):
         """update spec and push updates to remote git repo
 
-        :param branch:  target git branch
-        :param message: git commit message
-        :param update:  update files (git add update=True)
-        :param remote:  git remote, default to origin
-        :param add:     list of files to add
+        :param branch:       target git branch
+        :param message:      git commit message
+        :param update:       update files (git add update=True)
+        :param remote:       git remote, default to origin
+        :param add:          list of files to add
+        :param author_name:  author's git user name to be used on this commit
+        :param author_email: author's git user email to be used on this commit
+        :param secrets:      dict or SecretsStore with Git credentials e.g. secrets={"GIT_TOKEN": token}
         """
         repo = self.spec.repo
         if not repo:
             raise ValueError("git repo is not set/defined")
         self.save()
 
+        if author_name and author_email:
+            config = repo.config_writer()
+            config.set_value("user", "name", author_name)
+            config.set_value("user", "email", author_email)
+            config.release()
```

Comment:
As a user, I would expect that if I supply only 1 of them it would be set. You can change to:
```suggestion
        with repo.config_writer() as config:
            if author_name:
                    config.set_value("user", "name", author_name)
            if author_email:
                    config.set_value("user", "email", author_email)
```

Reply:
That's fair. I added your improvement.